src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.cs

```diff
@@ -49,7 +50,37 @@ public override DataFrameColumn Add(DataFrameColumn column, bool inPlace = false
         }
         public override DataFrameColumn Add<U>(U value, bool inPlace = false)
         {
-            return AddImplementation(value, inPlace);
+            switch (typeof(U))
```

Comment:
We should change this to `if (typeof(U) == typeof(bool))`.

1. It is simpler than this switch case.
2. The JIT can specialize the generic so the whole `if` statements go away when it is JIT'ing for a specific type.